pkg/registry/apis/dashboard/legacy/storage.go

```diff
@@ -194,7 +193,7 @@ func (a *dashboardSqlAccess) ListIterator(ctx context.Context, req *resource.Lis
 			_ = rows.Close()
 		}()
 	}
-	if err != nil {
+	if err == nil {
```

Comment:
^^^ the real error

The rest of the PR is cleanup and making the tests work better